src/stream.rs

```diff
@@ -411,326 +416,363 @@ mod tests {
 
     #[test]
     fn empty_read() {
-        let mut buf = RecvBuf::default();
-        assert_eq!(buf.len(), 0);
+        let mut recv = RecvBuf::default();
+        assert_eq!(recv.len(), 0);
+
+        let mut buf = vec![0; 32];
+        assert_eq!(buf.len(), 32);
```

Comment:
No need to test that `vec![]` works, we can just assume that the compiler is not broken :)

Same below in the other tests.